src/sentry/models/repository.py

```diff
@@ -93,6 +93,29 @@ def reset_pending_deletion_field_names(
         del self.config["pending_deletion_name"]
         return super().reset_pending_deletion_field_names(["config"])
 
+    def cascade_delete_on_hide(self) -> None:
```

Comment:
Doing this cleanup work synchronously in the endpoint could result in timeouts if a repository has a large number of pull requests and commits. Could we spawn a task to delete the repository relations?